### ISSUE TITLE
[Request] Remove OpenCV requirement from the core for 300MB

### ISSUE THREAD

**Issue by m-roberts** [pi-top/pi-top-Python-SDK#582]
### Problem or Possibility

Currently OpenCV is installed with the core of the SDK, but it appears to only be used by `ImageFunctions.convert`
Removing this dependency from the core would help to reduce install footprint for core functionality.

It is used in `pitop.processing` and `pitop.camera`, but - most importantly - it does the [heavy lifting](https://github.com/pi-top/pi-top-Python-SDK/blob/35d189c1a95dbc4e1dd6b4f55f987ea6c488a524/packages/miniscreen/pitop/miniscreen/oled/oled.py#L276) of the miniscreen's `display_image` function.

### Proposed Solution

An OpenCV image is unlikely to be passed in without the `cv2` dependency being installed, so it shouldn't even be necessary to invoke the `convert` function anyway.

Perhaps the `display_image` function could try to convert if the image format is PIL and raises an exception if it's an OpenCV image without the dependency installed?

### Alternative Solutions

According to [this](https://answers.opencv.org/question/219040/fastest-way-to-convert-bgr-rgb-aka-do-not-use-numpy-magic-tricks/), this operation should also work:
```python
x = x[...,::-1].copy()
```

However, this does not have OpenCV's speed benefit. Perhaps this can be used as a fallback option if OpenCV is not available?

**Comment by hramrach** [comment]
Yes, the requirement to install 3 different image processing frameworks for the ability to power off the device is kind of lame.

I suspect this is due to bad design of the python libraries. Why do you need to import ImageFunctions.convert?

Can't the conversion function be a method on each image object? Then you would call the method of whichever library provided the image, transparently.
